### DISCUSSION TITLE
Analytics - Item level extra properties

### DISCUSSION THREAD

**Discussion by WiltonGodinho** [invertase/react-native-firebase#4340] [Ideas] [answered]
HI @mikehardy i am using react-native-firebase/analytics-7.6.2. Is it possible to add extra properties to the ITEM obj like:
 - Price
 - Quantity

As per the documentation, we can just use the default price and quantity parameters for the ITEM level.

```
//A pair of socks
let item_socks = {
   item_id: 'SKU_789',
   item_name: 'ankle_socks',
   item_category: 'socks',
   item_variant: 'red',
   item_brand: 'Google',
   price: 5.99
};
```

source: https://firebase.google.com/docs/analytics/measure-ecommerce#javascript

Thank you so much for your great work.

**Comment by mikehardy** [comment, marked as answer]
Hi there! A PR to extend coverage of any of the APIs is always welcome. Philosophically the javascript API is what the "shape" of the API here should conform to (naming, parameters, return values) if at all possible, but under the covers of course we are constrained to the underlying firebase-ios-sdk + firebase-android-sdk APIs

I think these are them?
https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Event
https://firebase.google.com/docs/reference/ios/firebaseanalytics/api/reference/Classes/FIRAnalytics

so a PR that implemented the appropriate properties in the firebase-js-sdk style, bent into the correct underlying native SDK APIs would do the trick

Development on the library is done by cloning the repo then opening the "tests" app as if it were an app you were developing, with the e2e tests in the modules serving as the "feature" in the "app" you are building, and the native +JS code of the module all available via Xcode and Android Studio and VSCode to implement the feature

An example feature that's relatively easy to understand since it is small might show how the final changes end up #3814
